catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/fragment/MergeProjectNameTest.java

```diff
@@ -0,0 +1,166 @@
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.StartScript;
+import org.catrobat.catroid.content.bricks.SetXBrick;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.io.XstreamSerializer;
+import org.catrobat.catroid.testsuites.annotations.Cat;
+import org.catrobat.catroid.testsuites.annotations.Level;
+import org.catrobat.catroid.ui.ProjectListActivity;
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityTestRule;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.InstrumentationRegistry.getInstrumentation;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.isEnabled;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.ui.fragment.rvutils.RecyclerViewInteractionWrapper.onRecyclerView;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.not;
+
+@RunWith(AndroidJUnit4.class)
+public class MergeProjectNameTest {
+
+	@Rule
+	public BaseActivityTestRule<ProjectListActivity> baseActivityTestRule = new
+			BaseActivityTestRule<>(ProjectListActivity.class, true, false);
+
+	private String firstProjectName = "firstProjectName";
+	private String secondProjectName = "secondProjectName";
+	private String validName = "ValidName";
+	private String inValidName = "firstProjectName";
+
+	@Before
+	public void setUp() throws Exception {
+		createProject(firstProjectName);
+		createProject(secondProjectName);
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void mergeProjectsValidNameTests() {
+		openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+		onView(withText(R.string.merge)).perform(click());
+
+		onRecyclerView().atPosition(0)
+				.performCheckItem();
+
+		onRecyclerView().atPosition(1)
+				.performCheckItem();
+
+		onView(withId(R.id.confirm)).perform(click());
+
+		onView(allOf(withText(""), isDisplayed(), instanceOf(EditText.class)))
+				.perform(replaceText(validName));
+		closeSoftKeyboard();
```

Comment:
this "closeSoftKeyboard()" is not doing anything, because you are using the wrong import.
you are using android.support.test.espresso.action.ViewActions.closeSoftKeyboard, which is a ViewAction, so it has to be done in a perform. if you just state it outside, you create a new ViewAction object and not doing anything with it.

the correct one for this would be: android.support.test.espresso.Espresso.closeSoftKeyboard, which is a standalone action like "pressBack()".

so either:

import static android.support.test.espresso.action.ViewActions.closeSoftKeyboard;

... .perform(replaceText(validName), closeSoftKeyboard());

or:

import static android.support.test.espresso.Espresso.closeSoftKeyboard;
...
closeSoftKeyboard();